### DISCUSSION TITLE
Introspection with MongoDB

### DISCUSSION THREAD

**Discussion by erasebegin** [prisma/prisma#15836] [Q&A] [answered]
Hi,

I'm new to backend stuff. I've gone through the process of setting up a database on Atlas, and have followed the guide in the Prisma docs step by step: 
- I've used `prisma init` to add the schema and env to my project, 
- I've added the database url and password to the .env file,
- I've set the db provider as "mongodb"

But whenever I run `npx prisma db pull`, I get the following output:

```
Prisma schema loaded from prisma/schema.prisma
Environment variables loaded from .env
Datasource "db"

✖ Introspecting based on datasource defined in prisma/schema.prisma

Error: An invalid argument was provided: Database must be defined in the connection string in database URL
```

I thought maybe the issue was the .env not being loaded properly, but the terminal output says `variables loaded from .env`

Not sure what I'm doing wrong here.

**Comment by nurul3101** [comment, marked as answer]
Hello @erasebegin 👋 

Could you share the connection string which you are using? ( After obfuscating the username and password )
I am interested in seeing the connection string format.

**Reply by erasebegin** [reply]
`mongodb+srv://myteam:password@my-db.18hf8.mongodb.net/?retryWrites=true&w=majority`

I copied the whole string directly from Atlas and just added in the provided password

**Reply by janpio** [reply]
You do not have a database in your connection string. 
Add it before `?retryWrites` and Prisma will be able to introspect the content of the database you specify.

**Reply by erasebegin** [reply]
That solved the issue, thank you so much 🙏🙏🙏

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
